graphstorm-processing/graphstorm_processing/data_transformations/dist_transformations/dist_numerical_transformation.py

```diff
@@ -209,8 +237,166 @@ def gauss_transform(rank: pd.Series) -> pd.Series:
+    assemblers = [VectorAssembler(inputCols=[col], outputCol=col + "_vec") for col in cols]
+    scalers = [MinMaxScaler(inputCol=col + "_vec", outputCol=col + "_scaled") for col in cols]
+
+    vector_cols = [col + "_vec" for col in cols]
+    scaled_cols = [col + "_scaled" for col in cols]
+
+    pipeline = Pipeline(stages=assemblers + scalers)
+    # If transformation representation exists, use that to fit the pipeline,
+    # otherwise we just use the input DF
+    if original_max_vals and original_min_vals:
+        # Create a DF with just the min and the max value per column,
+        # and we use that to fit each scaler used to fit the pipeline.
+        # The dummy DF will have two rows and len(cols) columns.
+        # We use the first row to get the min values and the second row to get the max values
+        min_exprs = [F.lit(val).alias(col) for val, col in zip(original_min_vals, cols)]
+        max_exprs = [F.lit(val).alias(col) for val, col in zip(original_max_vals, cols)]
+
+        # We add a zipWithIndex column to distinguish the first and second rows
+        # We use a list comprehension with F.when() to set the values for each column.
+        # For the first row (where row number is 0), we use the values from original_min_vals,
+        # and for the second row, we use the values from original_max_vals.
+        # Example: minvals=[0, 4], maxvals=[100, 256], cols=["col1", "col2"] becomes
+        # |"col1"|"col2"|
+        # |0     |4     |
+        # |100   |256   |
+        dummy_df = (
+            input_df.limit(2)
+            .rdd.zipWithIndex()
+            .toDF()
+            .select(
+                *[
+                    F.when(F.col("_2") == 0, min_expr).otherwise(max_expr).alias(col_name)
+                    for min_expr, max_expr, col_name in zip(min_exprs, max_exprs, cols)
+                ]
+            )
+        )
+
+        # Fit a pipeline on just the dummy DF
+        scaler_pipeline = pipeline.fit(dummy_df)
+    else:
+        # Fit a pipeline on the entire input DF
+        scaler_pipeline = pipeline.fit(input_df)
+
+    # Transform the input DF
+    scaled_df = scaler_pipeline.transform(input_df).drop(*vector_cols).drop(*cols)
+
+    # F.col(scaled_col_name).getField('values'[0] get the first element of a SparseVector
+    # https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.ml.linalg.SparseVector.html
+    scaled_df = scaled_df.select(
+        *[
+            (vector_to_array(F.col(scaled_col_name), dtype=out_dtype)[0].alias(orig_col))
+            for scaled_col_name, orig_col in zip(scaled_cols, cols)
+        ]
+    )
+    # Spark pipelines arrange transformations in a list, ordered by stage/
+    # So here we have first all the VectorAssemblers for each feature, then all the
+    # MinMaxScalerModel for each feature. So we skip the first num_cols to
+    # get just the MinMaxScalerModels
+    min_max_models: list[MinMaxScalerModel] = scaler_pipeline.stages[len(cols) :]
```

Comment:
Do we need to check the types of scaler_pipeline.stages[len(cols) :] just in case the Spark implementation changes?